context/auth.tsx

```diff
@@ -58,18 +84,26 @@ export const AuthProvider = ({ children }) => {
                     .then((idToken) => {
                         axios
                             .post("login/", { id_token: idToken })
-                            .then((response) => {
-                                let newToken = response.data.token
-                                updateProfile(newToken)
-                                    .then((response) => {
-                                        setToken(newToken)
-                                        setProfile(response.data)
-                                        setLoading(false)
-                                    })
-                                    .catch((error) => {
-                                        setLoading(false)
-                                    })
-                            })
+                            .then(
+                                (
+                                    response: AxiosResponse<{ token: string }>
+                                ) => {
+                                    let newToken: string = response.data.token
+                                    updateProfile(newToken)
+                                        .then(
+                                            (
+                                                response: AxiosResponse<ProfileSerializer>
+                                            ) => {
```

Comment:
Since updateProfile returns `axios.get<profileType>(`profile/`)`, so, the response is handled here. Should I move the handling part inside updateProfile?

Reply:
Indeed that seems better according to the word `update` in the function name. Just make sure you stop loading after the function returns.

Reply:
done